contracts/SpokePool.sol

```diff
@@ -371,6 +374,9 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         SpokePoolInterface.RelayerRefundLeaf memory relayerRefundLeaf,
         bytes32[] memory proof
     ) public nonReentrant {
+        // This function potentially is expected to transfer WETH to the recipient so we should wrap all ETH.
+        _depositEthToWeth();
```

Comment:
follow on question: does it make sence to put this in the spoke Pool? would it not make _more_ sence to put this in the Optimism spoke pool only seeing this is only relevant on optimism?

Reply:
As discussed IRL i'll move to Optimism_SpokePool